pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/templates/scripts/health-monitor.tpl.sh

```diff
@@ -13,7 +13,7 @@ function kubelet_monitoring {
   }
 
   function restart_kubelet {
-    pkill -f "kubelet"
+    pkill "kubelet"
```

Comment:
By default `pkill` will kill any process which has `kubelet` as a substring in its name. If I understand the issue correctly we want to kill process with exact match to name `kubelet` for that you can use following suggestion.
```suggestion
    pkill -x "kubelet"
```
From `pkill` manual
```
-x                Require an exact match of the process name, or argument list if -f is given.  The default is to match any substring.
```

Reply:
Thanks for catching this.
I'd incorrectly assumed that `pkill -x` would need to exactly match the full path to the executable (ie. `argv[0]` of the command line that it was executed with), which is why I'd avoided it.
I've double-checked on one of our gardener nodes and using ` pkill -x "kubelet"` here is correct.
It matches against `/proc/$PID/comm` rather than `/proc/$PID/cmdline`:
```
$ k exec -it privileged-pod -- nsenter --all -t 1
# pgrep -x kubelet
20177
# cat /proc/20177/comm
kubelet
# cat /proc/20177/cmdline
/opt/bin/kubelet --bootstrap-kubeconfig=/var/lib/kubelet/kubeconfig-bootstrap ...
```